### PR TITLE
[JavaScript] standardize inheritance: SMS and smsbillables

### PR DESCRIPTION
**Pull request by lwyszomi**
Hi @orangejenny

I refactored next part of javascripts. JIRA ticket: https://dimagi-dev.atlassian.net/browse/HI-213

Please let me know if you have any questions.

Regards,
Łukasz

### CODE REVIEW

File: corehq/apps/sms/static/sms/js/settings.js [hunk 1]
```diff
@@ -9,185 +9,196 @@ hqDefine("sms/js/settings", [
     initialPageData,
     select2Handler
 ) {
+    $(function () {
```
Comment (orangejenny) [inline corehq/apps/sms/static/sms/js/settings.js]: Why did the definitions of `dayTimeWindow` and `settingsViewModel` move into the document ready handler? It isn't harmful, I'm just curious if there was a reason.

File: corehq/apps/smsbillables/static/smsbillables/js/smsbillables.rate_calc.js [hunk 2]
```diff
@@ -10,144 +10,145 @@ hqDefine("smsbillables/js/smsbillables.rate_calc", [
...
-        self.hasError = ko.observable(false);
-        self.noError = ko.computed(function () {
-            return ! self.hasError();
-        });
-        self.calculatingRate = ko.observable(false);
-        self.showRateInfo = ko.computed(function () {
-            return self.rate() && ! self.calculatingRate();
-        });
+    $(function () {
+        var smsRateCalculator = function (formData) {
```
[the remark this answers is not in the document]
Comment (orangejenny) [reply corehq/apps/smsbillables/static/smsbillables/js/smsbillables.rate_calc.js]: Line 151 is `var smsRateCalculator = smsRateCalculator({ ... })`

So `smsRateCalculator` is used, but it's creating another variable with the same name, which is confusing. Could you rename either the `smsRateCalculator` definition or the `smsRateCalculator` instance variable? There's a loose convention of adding "Model" to the end of the function definition (this kind of conflict has come up a few times when making the model definitions lowercase).

File: corehq/apps/smsbillables/static/smsbillables/js/smsbillables.rate_calc.js [hunk 3]
```diff
@@ -10,144 +10,145 @@ hqDefine("smsbillables/js/smsbillables.rate_calc", [
...
-        self.rate = ko.observable();
-        self.hasError = ko.observable(false);
-        self.noError = ko.computed(function () {
-            return ! self.hasError();
-        });
-        self.calculatingRate = ko.observable(false);
-        self.showRateInfo = ko.computed(function () {
-            return self.rate() && ! self.calculatingRate();
-        });
+    $(function () {
```
Comment (orangejenny) [inline corehq/apps/smsbillables/static/smsbillables/js/smsbillables.rate_calc.js]: Same question as above on moving model definitions into the document ready handler.

**Review by orangejenny — CHANGES_REQUESTED** [review]
Minor comments/questions. The `smsRateCalculator` rename is the only blocking comment.

File: corehq/apps/sms/static/sms/js/settings.js (hunk 1 above)
Comment (lwyszomi) [reply corehq/apps/sms/static/sms/js/settings.js]: @orangejenny Don't know why but if these definitions weren't in the document ready then they are not visible when we try to execute the constructor.

Comment (orangejenny) [reply corehq/apps/sms/static/sms/js/settings.js]: I don't follow - I'm thinking of a setup like this, where the models are outside the handler, but they'd still be accessible inside the handler:
```function dayTimeWindow (...) {
   ...
}

function settingsViewModel(...) {
   ...
}

$(function() {
   var settingsModel = settingsViewModel(...);
   ...
});```

Comment (orangejenny) [reply corehq/apps/sms/static/sms/js/settings.js]: But in any case, the code is fine either way, so I'm going to go ahead and merge this PR.

**Review by orangejenny — APPROVED** [review]

**Comment by orangejenny** [comment]
This PR says the build is pending, but it completed successfully: https://travis-ci.org/dimagi/commcare-hq/builds/478303239 . Merging.

**Pull request merged**